### PR TITLE
Fix failing 131_sort_segments_migrate_to_data_stream test

### LINKED ISSUE #77829: [CI] DataStreamsClientYamlTestSuiteIT test {p0=data_stream/131_sort_segments_migrate_to_data_stream/Test that datastream index segments are sorted on timestamp field desc after data stream migration} failing
Reproducible locally see https://gradle-enterprise.elastic.co/s/sw5e6hii655yc

**Build scan:**
https://gradle-enterprise.elastic.co/s/jusar2bzjazz4/tests/:x-pack:plugin:data-streams:qa:rest:yamlRestTest/org.elasticsearch.xpack.datastreams.DataStreamsClientYamlTestSuiteIT/test%20%7Bp0=data_stream%2F131_sort_segments_migrate_to_data_stream%2FTest%20that%20datastream%20index%20segments%20are%20sorted%20on%20timestamp%20field%20desc%20after%20data%20stream%20migration%7D

**Reproduction line:**
`./gradlew ':x-pack:plugin:data-streams:qa:rest:yamlRestTest' --tests "org.elasticsearch.xpack.datastreams.DataStreamsClientYamlTestSuiteIT.test {p0=data_stream/131_sort_segments_migrate_to_data_stream/Test that datastream index segments are sorted on timestamp field desc after data stream migration}" -Dtests.seed=AD4DA3B69D72AA83 -Dtests.locale=en-MT -Dtests.timezone=Europe/Nicosia -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.datastreams.DataStreamsClientYamlTestSuiteIT&tests.test=test%20%7Bp0%3Ddata_stream/131_sort_segments_migrate_to_data_stream/Test%20that%20datastream%20index%20segments%20are%20sorted%20on%20timestamp%20field%20desc%20after%20data%20stream%20migration%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [data_stream/131_sort_segments_migrate_to_data_stream:50]: Expected a list containing
0: expected "2021-08-01" but was "2021-08-02"

  at __randomizedtesting.SeedInfo.seed([AD4DA3B69D72AA83:25199C6C338EC77B]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:451)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:428)
  at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(null:-1)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

  Caused by: java.lang.AssertionError: Expected a list containing
  0: expected "2021-08-01" but was "2021-08-02"

    at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:79)
    at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:62)
    at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:89)
    at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:65)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:444)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:428)
    at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(null:-1)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:566)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
    at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
    at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
    at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
    at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
    at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
    at java.lang.Thread.run(Thread.java:834)

```

### PR DESCRIPTION
**Pull request by danhermann**
Fixes #77829

### CODE REVIEW

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/data_stream/131_sort_segments_migrate_to_data_stream.yml [hunk 1]
```diff
@@ -30,14 +30,14 @@
   - do:
       index:
         index: my_ds
-        body: { "foo": "bar1", "@timestamp": "2021-08-01" }
+        body: { "foo": "bar1", "@timestamp": "2021-08-21" }
```
Comment (danhermann) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/data_stream/131_sort_segments_migrate_to_data_stream.yml:33]: The test failure appears to be dependent on the locale (en-MT) and possibly the JDK version in the reproduction line. Changing the sample dates to ones in which the month and day cannot be validly swapped (i.e., is `2021-08-01` interpreted as `Jan 08 21` or `Aug 01 21`) eliminates the test failure, but I'm not sure if that's the correct fix or just masking the problem.

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

**Comment by mayya-sharipova** [comment]
I've checked failed tests that contained 2 shards in the response.
```js


1> [2021-09-22T01:02:01,660][INFO ][o.e.x.d.DataStreamsClientYamlTestSuiteIT] [test] Stash dump on test failure [{ |  
-- | --
  | 1>   "stash" : { |  
  | 1>     "body" : { |  
  | 1>       "took" : 10, |  
  | 1>       "timed_out" : false, |  
  | 1>       "_shards" : { |  
  | 1>         "total" : 2, |  
  | 1>         "successful" : 2, |  
  | 1>         "skipped" : 0, |  
  | 1>         "failed" : 0 |  
  | 1>       },

```

I've incorrectly assumed that `test_index1` will always have only 1 shard. So the way to fix to ensure `test_index1` is created with a single shard:

```diff
  - do:
      indices.create:
        index: test_index1
        body:
+         settings:
+           number_of_shards: 1
+           number_of_replicas: 0
          aliases:
            my_ds:
              is_write_index: true
```

**Comment by mayya-sharipova** [comment]
> > The test failure appears to be dependent on the locale (en-MT) and possibly the JDK version in the reproduction line. Changing the sample dates to ones in which the month and day cannot be validly swapped (i.e., is 2021-08-01 interpreted as Jan 08 21 or Aug 01 21)

Isn't that by default if a data format is not provided we always use `"strict_date_optional_time||epoch_millis"` so it should always be interpreted as `yyyy-MM-dd`?

**Comment by danhermann** [comment]
> @danhermann Thanks for looking into this issue.
> 
> > The test failure appears to be dependent on the locale (en-MT) and possibly the JDK version in the reproduction line. Changing the sample dates to ones in which the month and day cannot be validly swapped (i.e., is 2021-08-01 interpreted as Jan 08 21 or Aug 01 21)
> 
> Isn't that by default if a data format is not provided we always use `"strict_date_optional_time||epoch_millis"` so it should always be interpreted as `yyyy-MM-dd`?

Yes, you're right about that. I assumed that running with the same test seed would have resulted in the same number of shards for the index, but since I could reproduce the test failure only when the locale and JDK version were specified, I started looking for locale-specific items.

**Review by mayya-sharipova — APPROVED** [review]
Thanks for fixing the issue.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Pull request merged**